fsharp-kestrel-sample/src/Program.fs

```diff
@@ -1,55 +1,45 @@
-﻿namespace Sample
-
-// Adapted from https://github.com/Krzysztof-Cieslak/KestrelFSharp.git
-module Main = 
-    open System
-    open System.IO
-    open System.Text
-    open System.Threading.Tasks
-    open Microsoft.AspNetCore.Builder
-    open Microsoft.AspNetCore.Http
-    open Microsoft.AspNetCore.Http.Extensions
-    open Microsoft.AspNetCore.Hosting
-    open Serilog
-    open Serilog.Configuration
-    open Microsoft.Extensions.Logging
-    open Microsoft.Extensions.Configuration
-    open Microsoft.AspNetCore.Builder
-
-    type Startup (env: IHostingEnvironment) =
-        do
-            Log.Logger <- LoggerConfiguration()
-                            .MinimumLevel.Verbose()
-                            .WriteTo.LiterateConsole()
-                            .Enrich.FromLogContext() 
-                            .CreateLogger();
-
-        member this.Configuration
-            with get() = Microsoft.Extensions.Configuration.Memory.MemoryConfigurationSource(
-                            InitialData = (dict [ "Logging:IncludeScopes", "False"
-                                                  "Logging:LogLevel:System", "Warning"
-                                                  "Logging:LogLevel:Default", "Debug"
-                                                  "Logging:LogLevel:Microsoft", "Warning" ]))
-
-        member this.Configure (app : IApplicationBuilder, env: IHostingEnvironment, loggerFactory: ILoggerFactory) =
-            loggerFactory.AddSerilog() |> ignore
-
-            app.Run (fun ctx ->
-                let log = ctx.RequestServices.GetService(typeof<ILogger<unit>>) :?> ILogger<unit>
-                log.LogDebug("got request at {@requestUrl}", ctx.Request.GetEncodedUrl())
-                
-                sprintf "Hello from Serilog ASP.NET Core F# Sample!"
-                |> ctx.Response.WriteAsync
-            )
- 
+namespace FSharpKestrelSample
+
+open System.IO
+open Microsoft.AspNetCore
+open Microsoft.AspNetCore.Hosting
+open Microsoft.Extensions.Configuration
+open Serilog
+
+module Program =
+    let successExitCode = 0
+    let failureExitCode = 1
+
+    let Configuration =
+        ConfigurationBuilder()
+            .SetBasePath(Directory.GetCurrentDirectory())
+            .AddJsonFile("appsettings.json", optional=true, reloadOnChange=true)
+            .Build() :> IConfiguration
+
+    let CreateWebHostBuilder args =
+        WebHost
+            .CreateDefaultBuilder(args)
+            .UseSerilog()
+            .UseStartup<Startup>();
+
     [<EntryPoint>]
-    let main argv = 
-        WebHostBuilder()
-            .UseKestrel()
-            .UseContentRoot(Directory.GetCurrentDirectory())
-            .UseUrls("http://0.0.0.0:5001")
-            .UseStartup<Startup>()
-            .Build()
-            .Run()
-        
-        0 // return an integer exit code
\ No newline at end of file
+    let main args =
+
+        Log.Logger <- LoggerConfiguration()
```

Comment:
maybe `use` the Logger and assign to Log.Logger after ?

Reply:
The goal was to be basically the same [the C# recommended pattern](https://github.com/serilog/serilog-docker/blob/master/web-sample/src/Program.cs#L33), unless there’s a great reason to deviate that I have considered yet?

Reply:
A try-catch-finally sure looks nice ;) I guess as general advice, assigning to `Log.Logger` and then doing `Log.CloseAndFlush` does make the most sense. (My comment as in reaction to all the nesting - normally I put one liner `with` or `finally` things on one line and/or go out of my way to use `use`, but writing it as clearly as possible including the nesting is probably no harm)

Reply:
Thanks @bartelink